laterpay/application/Compatibility/Migrate.php

```diff
@@ -0,0 +1,317 @@
+            foreach ( $subscriptions as $subscription ) {
+                $count ++;
+                $args = array(
+                    'post_type'      => 'lp_subscription',
+                    'no_found_rows'  => true,
+                    'posts_per_page' => 1,
+                    'fields'         => 'ids',
+                    // Meta query is required for post id.
+                    'meta_key'       => '_lp_id', // phpcs:ignore
+                    'meta_value'     => $subscription->id, // phpcs:ignore
+                );
+
+                $result = new WP_Query( $args );
+                if ( ! $result->have_posts() ) {
+
+                    $data = array(
+                        'ID'              => '',
+                        'counter_id'      => (int) $subscription->id,
+                        'duration'        => (int) $subscription->duration,
+                        'period'          => (int) $subscription->period,
+                        'access_to'       => (int) $subscription->access_to,
+                        'access_category' => (int) $subscription->access_category,
+                        'price'           => (float) $subscription->price,
+                        'title'           => $subscription->title,
+                        'description'     => $subscription->description,
+                    );
+
+                    $success = $laterpay_subscription->update_subscription( $data, true );
+                    if ( $count === $size ) {
+                        update_option( 'lp_sub_count', absint( $subscription->id ) );
+                    }
+
+                    if ( 1 === (int) $subscription->is_deleted ) {
+                        $laterpay_subscription->delete_subscription_by_id( $success['id'] );
+                    }
+                }
+            }
+        }
+
+        return $offset + count( $subscriptions );
+    }
+
+    /**
+     * Migrate term table to cpt.
+     *
+     * @param int $offset migration starting id of term.
+     *
+     * @return int new offset.
+     */
+    public function migrate_terms_table( $offset ) {
+        global $wpdb;
+
+        $term_table     = $wpdb->prefix . 'laterpay_terms_price';
+        $sql            = "SELECT * FROM {$term_table} limit %d, %d ";
+        $terms          = $wpdb->get_results( $wpdb->prepare( $sql, $offset, 10 ) );
+        $laterpay_terms = LaterPay_Model_CategoryPriceWP::get_instance( true );
+
+        if ( ! empty( $terms ) ) {
+
+            foreach ( $terms as $term ) {
+                $term_exist = get_term_meta( $term->term_id, '_lp_revenue_model', true );
+
+                if ( empty( $term_exist ) ) {
+                    $laterpay_terms->set_category_price( $term->term_id, $term->price, $term->revenue_model );
+                }
+            }
+
+        }
+
+        return $offset + count( $terms );
+    }
+
+    /**
+     *  Drop laterpay custom tables.
+     */
+    public function drop_custom_tables() {
```

Comment:
This could be a very scary ajax endpoint, can we check that the user is logged in and add a capability check too?

Reply:
@tomjn added in in https://github.com/rtCamp/laterpay-wordpress-plugin/pull/1/commits/da5dd0ecf4b6d3336aa20a95ee64d9f8c329e089